File: C#/OOP/OOP/BMW.cs
using System;

namespace OOP {
    class BMW : Car {
        public BMW() : base() { }

        public BMW(string ownerName, FuelType fuelType, Colour colour)
            : base(ownerName, fuelType, colour) { }

        public bool IsSportModeOn { get; set; }

        public override void GetSpecialFeature()
        {
            Console.WriteLine("I have sport mode.");
        }

        public override void Drive() {
            Console.WriteLine("Driving BMW");
        }
    }
}

File: Java/src/BMW.java
public class BMW extends Car{

    private boolean sportModeOn;

    public BMW() {
        super();
    }

    public BMW(String ownerName, FuelType fuelType, Colour colour) {
        super(ownerName, fuelType, colour);
    }

    public boolean isSportModeOn() {
        return sportModeOn;
    }

    public void setSportModeOn(boolean sportModeOn) {
        this.sportModeOn = sportModeOn;
    }

    @Override
    public void getSpecialFeature() {
        System.out.println("I have sports mode.");
    }

    @Override
    public void drive() {
        System.out.println("Driving BMW");
    }
}
